core/common/test/LocalDateRangeTest.kt

```diff
@@ -0,0 +1,283 @@
+/*
+ * Copyright 2019-2022 JetBrains s.r.o.
+ * Use of this source code is governed by the Apache 2.0 License that can be found in the LICENSE.txt file.
+ */
+
+package kotlinx.datetime.test
+
+import kotlinx.datetime.*
+import kotlinx.datetime.internal.clampToInt
+import kotlin.random.Random
+import kotlin.random.nextLong
+import kotlin.test.*
+
+class LocalDateRangeTest {
```

Comment:
Could you also add tests for equals/hashCode contract of ranges/progressions?
E.g. that a progression with the step of 1 day is equal to a range. Or that a progression with the step of 7 days is equal to a progression with the step of a week.